src/main/java/core/basesyntax/strategy/BalanceOperationHandlerImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperationHandlerImpl implements OperationHandler {
+
+    @Override
+    public void handleOperation(FruitTransaction transaction) {
+        String fruitName = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+        try {
+            if (quantity >= 0) {
+                FruitStorage.getFruits().put(fruitName, transaction);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("quantity must be positive value");
+        }
```

Comment:
you won't catch any exception (nothing could generate it). Just add a "guard"check - if' it's less than 0 - thrown exception. After that check, a line 14 should go. We don't need to wrap it into anything, because we won't reach it if exception is thrown